processor/memorylimiter/memorylimiter.go

```diff
@@ -100,6 +107,24 @@ func newMemoryLimiter(logger *zap.Logger, cfg *Config) (*memoryLimiter, error) {
 	return ml, nil
 }
 
+func getDecision(cfg *Config, logger *zap.Logger) (dropDecision, error) {
+	memAllocLimit := uint64(cfg.MemoryLimitMiB) * mibBytes
+	memSpikeLimit := uint64(cfg.MemorySpikeLimitMiB) * mibBytes
+	if cfg.MemoryLimitMiB != 0 {
+		return newFixedDecision(memAllocLimit, memSpikeLimit)
+	}
+	totalMemory, err := getMemoryFn()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get total memory, use fixed memory settings (limit_mib): %w", err)
+	}
+	decision, err := newPercentrageDecision(totalMemory, int64(cfg.MemoryLimitPercentage), int64(cfg.MemorySpikePercentage))
+	logger.Info("Using percentage settings for memory limiting",
+		zap.Int64("total_memory", totalMemory),
+		zap.Uint64("limit_mib", decision.memAllocLimit),
```

Comment:
`decision` can be `nil` here in case of error, even if there is no code path returning error right now it can become a crashing bug in a future change.